CONTRIBUTING.md

```diff
@@ -36,6 +36,217 @@ sudo ifconfig lo0 alias 127.0.0.3 up
 sudo ifconfig lo0 alias 127.0.0.4 up
 ```
 
+## Testing in Crawlee with vitest
+
+There are a few small differences between how testing in jest and vitest works. Mostly, they relate to what to do, and not do anymore.
+
+### Configuration file for tests created in the package they are for
+
+You will need to use this tsconfig.json in the `test` folder in the package (say, if you were adding a test to `packages/core` and there wasn't a `tsconfig.json` file already there)
+
+```json
+{
+ "extends": "../../../tsconfig.json",
+ "include": [
+  "**/*",
+  "../../**/*"
+ ],
+ "compilerOptions": {
+  "types": ["vitest/globals"]
+ }
+}
+```
+
+### Mocking modules
+
+Mocks are pretty much the same when it comes to jest vs vitest. One crucial difference is that you no longer need to unmock modules in an afterAll block, as they are mocked per test file.
+
+#### Previous
+
+```ts
+jest.mock("node:os", () => {
+ const original: typeof import("node:os") = jest.requireActual("node:os");
+ return {
+  ...original,
+  platform: () => "darwin",
+  freemem: jest.fn(),
+ };
+});
+
+afterAll(() => {
+  jest.unmock("node:os");
+});
+```
+
+#### Now
+
+```ts
+vitest.mock("node:os", async (importActual) => {
+  const original = await importActual<typeof import("node:os")>();
+ return {
+  ...original,
+  platform: () => "darwin",
+  freemem: jest.fn(),
+ };
+});
+```
+
+### Mocking based on imports
+
+Given the following two samples:
+
+#### 1
+
+```ts
+import os from "node:os";
+
+console.log(os.platform());
+```
+
+#### 2
+
+```ts
+import { platform } from "node:os";
+
+console.log(platform());
+```
+
+You will need to mock the module based on how you import it in the source code. This means, if you will import the default export, you will need to add a `default` property to the mocked object. Otherwise, you will need to mock the module as is.
+
+So, for example 1:
+
+```ts
+vitest.mock("node:os", async (importActual) => {
+ const original = await importActual<typeof import("node:os") & { default: typeof import("node:os") }>();
+
+ const platformMock = () => "darwin";
+ const freememMock = vitest.fn();
+
+ return {
+  ...original,
+  platform: platformMock,
+  freemem: freememMock,
+  // Specifically, you'll need to add this v block
```

Comment:
what is "v block"?

Reply:
It's meant to signal the block below the line, but that could've been done clearer